Logger/BaseEntity.cs

```diff
@@ -0,0 +1,19 @@
+
+namespace Logger;
+public abstract record class BaseEntity : IEntity
+
+    {
+        //because this is an internal ID used by developers and not the same as an employee or student id it is implemented explicitly
+         private Guid _InternalId = Guid.NewGuid();    
+         Guid IEntity.Id { get => _InternalId; init => _InternalId = value; }

```

Comment:
Why is a backing field needed? Would something like this work?

Additionally, I would expect the ID maybe wants to be set by the parent? But it could probably be argued either way.

```suggestion
        //because this is an internal ID used by developers and not the same as an employee or student id it is implemented explicitly
         Guid IEntity.Id { get;  } = Guid.NewGuid();    
```